core/src/main/java/org/apache/iceberg/hadoop/HadoopInputFile.java

```diff
@@ -224,9 +224,13 @@ public String location() {
   @Override
   public boolean exists() {
     try {
-      return fs.exists(path);
-    } catch (IOException e) {
-      throw new RuntimeIOException(e, "Failed to check existence for file: %s", path);
+      return lazyStat() != null;
+    } catch (RuntimeIOException re) {
+      if (re.getCause() instanceof FileNotFoundException) {
```

Comment:
Question, what do you think, about making lazyStat() catch FileNotFoundException specifically and throwing Iceberg's NotFoundException in that case?  Then here we can catch directly NotFoundException?

Reply:
I think it makes sense :) , made the corresponding changes !